projects/core/src/lib/components/drawing/drawing/drawing.component.ts

```diff
@@ -73,14 +102,34 @@ export class DrawingComponent implements OnInit, OnDestroy {
     this.destroyed.complete();
   }
 
-  public featureStyleUpdates(style: DrawingFeatureStyleModel) {
-    if (this.selectedFeature) {
-      this.store$.dispatch(updateDrawingFeatureStyle({ fid: this.selectedFeature.__fid, style }));
+  public onDrawingAdded($event: DrawingToolEvent) {
+    if (!this.activeTool) {
+      return;
     }
-    DrawingHelper.updateDefaultStyle({
-      ...style,
-      label: '',
-    });
+    this.store$.dispatch(addFeature({
+      feature: DrawingHelper.getFeature(this.activeTool, $event),
+      selectFeature: true,
+    }));
+  }
+
+  public onActiveToolChanged($event: DrawingFeatureTypeEnum | null) {
+    this.activeTool = $event;
+    this.store$.dispatch(setSelectedDrawingStyle({ drawingType: $event }));
+  }
+
+  public onFeatureSelected(feature: FeatureModel | null) {
+    this.store$.dispatch(setSelectedFeature({ fid: feature?.__fid || null }));
+  }
+
+  public onFeatureGeometryModified(geometry: string) {
+    this.store$.dispatch(updateSelectedDrawingFeatureGeometry({ geometry }));
+  }
+
+
+  @ViewChild(MapDrawingButtonsComponent) private mapDrawingButtonsComponent: MapDrawingButtonsComponent | undefined;
+
+  public enableSelectAndModify() {
+    this.mapDrawingButtonsComponent?.enableSelectAndModify();
```

Comment:
Not sure if we should do this. Dont think we execute child component methods anywhere else. See if we can come up with something else, otherwise, if it works, it works